tests/acceptance/04_examples/syntax/check_syntax.cf

```diff
@@ -0,0 +1,166 @@
+#######################################################
+#
+# Check the syntax of all examples in examples directory
+#
+# Is this test failing? If you added an example, make sure you add the
+# necessary bodies/bundles to mock_stdlib.cf
+# To find out which example is failing, run in tests/acceptance:
+# $ ./testall 04_examples/syntax/check_syntax.cf ; less test.log
+#
+#######################################################
+
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+}
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "basedir" string => "$(G.cwd)/../../examples";
+    "all_examples" slist => findfiles("$(basedir)/*.cf");
+    "stdlib" string => "$(G.testdir)/inputs/lib/stdlib.cf";
+    "mock" string => "$(this.promise_dirname)/../mock_stdlib.cf";
+
+  methods:
+    "stdlib_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/stdlib.cf");
+    "paths_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/paths.cf");
+    "users_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/users.cf");
+    "services_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/services.cf");
+    "packages_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/packages.cf");
+    "commands_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/commands.cf");
+    "files_symlink"
+      usebundle => lib_maybe_symlink(
+        "$(mock)", "$(G.testdir)/inputs/lib/files.cf");
+
+    DEBUG_LOTS::
+      "$(this.bundle): will consider example $(all_examples)";
+}
+
+bundle agent lib_maybe_symlink(src, dst)
+# Symlink a file, only if dst does not exist
+# If dst already exists, assume it's correct (testing with masterfiles)
+{
+  classes:
+    "file_missing"
+      if => not(fileexists("$(dst)"));
+  methods:
+    file_missing::
+      "symlink"
+        usebundle => symlink("$(src)", "$(dst)");
```

Comment:
You could just do `if => ...` on the `methods` promise, right?

Reply:
Yes, but I think this is the more "normal" way to do it. Define classes to make decisions using class guards. `if` (previously `ifvarclass`) has, at least traditionally, 1 specific purpose: To be able to use variables in class expressions.